crates/manifest/src/normalize.rs

```diff
@@ -0,0 +1,72 @@
+//! Manifest normalization functions.
+
+use std::collections::HashSet;
+
+use crate::schema::v2::{self, AppManifest, ComponentSpec, KebabId};
+
+/// Extracts each `ComponentSpec::Inline` into `AppManifest::component`s,
+/// replacing it with a `ComponentSpec::Reference`.
+pub fn normalize_inline_components(manifest: &mut AppManifest) {
```

Comment:
Perhaps I'm confused by the name `ComponentSpec`, but I'm confused by what this function and `normalize_trigger_ids` actually do a high-level (I get that they set some missing values but why?).

Reply:
It makes code that consumes manifests simpler by not dealing with the flexibility that these normalize.